components/automate-cli/pkg/verifyserver/services/softwareversionservice/softwareversionservice.go

```diff
@@ -0,0 +1,203 @@
+		}
+	}
+	return &models.Checks{
+		Title:         cmdName + AVAILABILITY,
+		Passed:        true,
+		SuccessMsg:    cmdName + " is available",
+		ErrorMsg:      "",
+		ResolutionMsg: "",
+	}
+}
+
+func (sv *SoftwareVersionService) checkOsVersion(osFilepath string) (*models.Checks, error) {
+	osVersions := map[string][]string{
+		"Red Hat":      {"7,8,9"},
+		"Ubuntu":       {"16.04.x", "18.04.x", "20.04.x", "22.04.x"},
+		"Centos":       {"7"},
+		"Amazon Linux": {"2"},
+		"SUSE Linux":   {"12"},
+		"Debian":       {"9", "10", "11", "12"},
+	}
+	checkResponse := models.Checks{}
+	var osName, osVersion, err = getosutils.GetOsVersion(osFilepath)
+	sv.logger.Debug("Got the OS Version = ", osVersion)
+	sv.logger.Debug("Got the OS Name = ", osName)
+	if err != nil {
+		sv.logger.Error("Enable to get OS Version as file on path doesnot exit = ", err)
+		return &models.Checks{
+			Title:         LINUX_VERSION_CHECK,
+			Passed:        false,
+			SuccessMsg:    "",
+			ErrorMsg:      "Its not feasible to determine the Operating system version",
+			ResolutionMsg: "Please run system on the supported platforms.",
+		}, nil
+	}
+	for key := range osVersions {
+		if strings.Contains(osName, key) {
+			correctVersion := sv.checkOs(osVersions, osVersion, key)
+			if correctVersion {
+				checkResponse = models.Checks{
+					Title:         LINUX_VERSION_CHECK,
+					Passed:        true,
+					SuccessMsg:    key + " version is " + osVersion,
+					ErrorMsg:      "",
+					ResolutionMsg: "",
+				}
+				break
+			}
+			checkResponse = models.Checks{
+				Title:         LINUX_VERSION_CHECK,
+				Passed:        false,
+				SuccessMsg:    "",
+				ErrorMsg:      key + " version is not supported by automate",
+				ResolutionMsg: ENSURE + key + " correct version is installed on the node",
+			}
+			break
+		}
+		checkResponse = models.Checks{
+			Title:         LINUX_VERSION_CHECK,
+			Passed:        false,
+			SuccessMsg:    "",
+			ErrorMsg:      osName + " version is not supported by automate",
+			ResolutionMsg: ENSURE + osName + " correct version is installed on the node",
+		}
+	}
+	return &checkResponse, nil
+}
+
+func (sv *SoftwareVersionService) checkOs(osVersions map[string][]string, osVersion string, osName string) bool {
+	correctVersion := false
+	if osName == UBUNTU {
+		split := strings.Split(osVersion, ".")
+		checkVersion := split[0] + "." + split[1]
+		re := regexp.MustCompile(checkVersion)
+
+		for _, str := range osVersions[osName] {
+			if re.MatchString(str) {
+				correctVersion = true
+				break
+			}
+		}
+	} else if osName == DEBIAN {
+		for _, str := range osVersions[osName] {
+			if str == osVersion {
+				correctVersion = true
+				break
+			}
+		}
+	} else if osName == RED_HAT {
+		checkVersion, _ := strconv.ParseFloat(osVersion, 64)
+		if checkVersion >= RED_HAT_SUPPORTED_VERSION {
+			correctVersion = true
+		}
+	} else {
+		if osVersions[osName][0] == osVersion {
+			correctVersion = true
+		}
+	}
+	return correctVersion
+}
```

Comment:
This function can be simplified 
for example like: below is how i meant, please add the other conditions as well, 
you can make it a switch case as well
```
func (sv *SoftwareVersionService) checkOs(osVersions map[string][]string, osVersion string, osName string) bool {
	if supportedVersions, ok := osVersions[osName]; ok {
		for _, version := range supportedVersions {
			if osName == UBUNTU {
				split := strings.Split(osVersion, ".")
				checkVersion := split[0] + "." + split[1]
				if strings.HasPrefix(version, checkVersion) {
					return true
				}
			}
		}
	}

	return false
}
```

Reply:
Done